planer-uwr-webapp/imports/utils.ts

```diff
@@ -12,23 +12,44 @@ export function isMeteorError(
   );
 }
 
-export const shortNameByType: Record<number, string | undefined> = {
-  '5': 'I1',
-  '6': 'I2',
-  '7': 'I.inż',
-  '8': 'O1',
-  '9': 'O2',
-  '10': 'O3',
-  '13': 'P',
-  '14': 'S',
-  '15': 'N',
-  '17': 'L',
-  '35': 'Inny',
-  '36': 'K1',
-  '37': 'K2',
-  '38': 'I2.T',
-  '39': 'I2.Z',
-  '40': 'K.inż',
-  '42': 'HS',
-  '43': 'M',
+export type Color = [number, number, number];
+
+export interface CourseType {
+  name: string;
+  fullName: string;
+  color: Color;
+}
+
+export const courseTypeById: Record<number, CourseType> = {
+  '5': { name: 'I1', color: [0, 100, 240], fullName: 'Informatyczny 1' },
+  '6': { name: 'I2', color: [0, 100, 240], fullName: 'Informatyczny 2' },
+  '7': { name: 'I.inż', color: [90, 40, 255], fullName: 'Informatyczny Inynierski' },
+  '8': { name: 'O1', color: [200, 0, 0], fullName: 'Obowiązkowy 1' },
+  '9': { name: 'O2', color: [200, 0, 0], fullName: 'Obowiązkowy 2' },
+  '10': { name: 'O3', color: [200, 0, 0], fullName: 'Obowiązkowy 3' },
+  '13': { name: 'P', color: [0, 170, 90], fullName: 'Projekt' },
+  '14': { name: 'S', color: [0, 130, 130], fullName: 'Seminarium' },
+  '15': { name: 'N', color: [120, 150, 20], fullName: 'Nieinformatyczny' },
+  '17': { name: 'L', color: [120, 150, 20], fullName: 'Laboratorium' },
+  '35': { name: 'Inny', color: [125, 125, 125], fullName: 'Inny' },
+  '36': { name: 'K1', color: [130, 30, 200], fullName: 'Kurs Podstawowy' },
+  '37': { name: 'K2', color: [130, 30, 200], fullName: 'Kurs Zaawansowany' },
+  '38': { name: 'I2.T', color: [0, 100, 240], fullName: 'Informatyczny 2 - Teoria' },
+  '39': { name: 'I2.Z', color: [0, 100, 240], fullName: 'Informatyczny 2 - Zastosowania' },
+  '40': { name: 'K.inż', color: [180, 30, 200], fullName: 'Kurs Inżynierski' },
+  '41': { name: 'PS', color: [0, 130, 130], fullName: 'Proseminarium' },
+  '42': { name: 'HS', color: [150, 120, 40], fullName: 'Humanistyczno-społeczny' },
+  '43': { name: 'M', color: [255, 135, 0], fullName: 'Matematyczny' }
 };
+
+export const generalCourseTypes: Record<number, CourseType> = {
+  '1': { name: 'O', color: [200, 0, 0], fullName: 'Wszystkie Obowiązkowe' },
+  '2': { name: 'I', color: [0, 100, 240], fullName: 'Wszsytkie Informatyczne' },
+  '3': { name: 'K', color: [130, 30, 200], fullName: 'Wszystkie Kursy' },
+  '4': { name: 'P', color: [0, 170, 90], fullName: 'Wszystkie Projekty' },
+}
+
+export const getTextColor = (bg: Color) => {
+  var yiq = ((bg[0]*299)+(bg[1]*587)+(bg[2]*114)) / 1000;
+  return (yiq >= 128) ? 'black' : 'white';  
```

Comment:
Can we get a little explanation about what's going on here?

Reply:
@Ph0enixKM

Reply:
This is just a function that determines if color is dark or light. If you prefer to use something different then please let me know. As for now this solution is in my opinion sufficient for this kind of project. The reason I've implemented it myself is because I wanted to use some library that is written in TypeScript instead of the existing pure JavaScript packages available on NPM and on the other hand I didn't want to write another small library from scratch.